### PR TITLE
Webpack 2. HMR using webpack-dev-server

### PR DESCRIPTION
**Pull request by chrisnojima**
We used to use our own express server and use a middleware that wrapped HMR stuff. Instead we use the web server that webpack themselves makes

### CODE REVIEW

File: shared/desktop/server.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 // @flow
 // Builds our code, serves changes if NO_SERVER is false
-const express = require('express')
```
Comment (chrisnojima) [inline shared/desktop/server.js]: get rid of our own server

File: shared/desktop/webpack.common.js [hunk 2]
```diff
@@ -16,14 +16,16 @@ const mockRule = {
...
 
 const isHot = getenv.boolish('HOT', false)
-const noSourceMaps = getenv.boolish('NO_SOURCE_MAPS', true)
+const noSourceMaps = getenv.boolish('NO_SOURCE_MAPS', false)
 
-const HMRUrl = 'webpack-hot-middleware/client?path=http://localhost:4000/__webpack_hmr'
-const RHLPatch = 'react-hot-loader/patch'
+const HMRPrefix = [
+  'react-hot-loader/patch',
+  'webpack-dev-server/client?http://localhost:4000',
```
Comment (chrisnojima) [inline shared/desktop/webpack.common.js]: use webpacks dev server

File: shared/desktop/webpack.config.development.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ const makePlugins = () => {
   })
 
   const dashboardPlugin = !noServer && [new DashboardPlugin()]
-  const hmrPlugin = isHot && [new webpack.HotModuleReplacementPlugin()]
+  const hmrPlugin = isHot && [new webpack.HotModuleReplacementPlugin(), new webpack.NamedModulesPlugin()]
```
Comment (chrisnojima) [inline shared/desktop/webpack.config.development.js]: namedmodules helps decode the hmr names

File: shared/desktop/webpack.config.main-thread-only.js [hunk 4]
```diff
@@ -20,7 +20,7 @@ const makePlugins = () => {
 const config = {
   ...baseConfig,
   bail: true,
-  devtool: noSourceMaps ? undefined : 'inline-eval-cheap-source-map',
+  devtool: undefined,
```
Comment (chrisnojima) [inline shared/desktop/webpack.config.main-thread-only.js]: don't ever use sourcemaps in main thread

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready to look at. if its fine you can merge it as i'm out fri/mon

**Comment by keybase-ci-visdiff** [comment]
The commits 2b5f2cf6bbc6...4dff1f1dc486 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Tracker-New_user_Scroll_1** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496368504104-2b5f2cf6bbc6-4dff1f1dc486-linux/Tracker-New_user_Scroll_1-2b5f2cf6bbc6.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496368504104-2b5f2cf6bbc6-4dff1f1dc486-linux/Tracker-New_user_Scroll_1-4dff1f1dc486.png)  
   Tracker-New_user_Scroll_1 comparison
</summary>

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Pull request merged**